### PR TITLE
fix(49544): Delimiting accessors in interface with comma causes parser error

### LINKED ISSUE #49544: Delimiting accessors in interface with comma causes parser error
# Bug Report
```ts
interface I {
  get x(): number,
  //             ^-- '{' expected.(1005)
}
```
Usually, one is free to use either only a newline, a semicolon, or a comma, when delimiting elements. However, for accessors in interfaces, commas cause a parser error.

### 🔎 Search Terms

interface accessor get set comma

### 🕗 Version & Regression Information

- This seems to be the behavior since the feature was introduced in TS 4.3.

### ⏯ Playground Link

[Playground link on nightly](https://www.typescriptlang.org/play?ts=4.8.0-dev.20220614#code/JYOwLgpgTgZghgYwgAgJLIN4FgBQzkDmEYyAHgBQCUAXMiAK4C2ARtADS74D0X+f--AHoBaYcgDkGccgikADhASQAJgDpyARgAMWgKyVcAXyA)

### 🙁 Actual behavior

Using a comma causes an error.

### 🙂 Expected behavior

Using a comma causes no error.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #49544

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -6673,11 +6673,16 @@ namespace ts {
         }
 
         function parseFunctionBlockOrSemicolon(flags: SignatureFlags, diagnosticMessage?: DiagnosticMessage): Block | undefined {
-            if (token() !== SyntaxKind.OpenBraceToken && canParseSemicolon()) {
-                parseSemicolon();
-                return;
+            if (token() !== SyntaxKind.OpenBraceToken) {
+                if (flags & SignatureFlags.Type) {
+                    parseTypeMemberSemicolon();
```
Comment (sandersn) [inline src/compiler/parser.ts:6678]: does this change parsing of other function-like declarations, particularly methods? Or, to reverse the question: how did method signature parsing work before? Does it not use parseFunctionBlockOrSemicolon? If not, why does parseAccessDeclaration?

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good, except I want to know how method declaration parsing allowed `,` without this fix.

File: src/compiler/parser.ts (hunk 1 above)
Comment (jakebailey) [reply src/compiler/parser.ts:6678]: `parseMethodDeclaration` calls `parseFunctionBlockOrSemicolon`, if that's what you mean.

Comment (a-tarasyuk) [reply src/compiler/parser.ts:6678]: `parseTypeMember` doesn't parse `,`, `;` (`parseTypeMemberSemicolon`) tokens after accessors

https://github.com/microsoft/TypeScript/blob/569cdf1b074dd7462c53102e0f69bfc9863a12d7/src/compiler/parser.ts#L3633-L3635

`parseAccessorDeclaration` is general method for type members/literals/class members for. Parsing `,`, `;` should be allowed for type members.`parseFunctionBlockOrSemicolon` has signature flags that include the `Type` option. I used it to allow parsing `,`, `;`. In other cases, the behavior stays the same. (`parseFunctionBlockOrSemicolon(SignatureFlags.None)`) Does that make sense?

**Pull request merged**